crates/native_std/src/lib.rs

```diff
@@ -163,13 +163,15 @@ pub fn git_revision_full() -> Option<String> {
 pub struct AmbientVersion {
     pub version: String,
     pub revision: String,
+    pub revision_full: String,
 }
 
 impl Default for AmbientVersion {
     fn default() -> Self {
         Self {
             version: env!("CARGO_PKG_VERSION").to_string(),
-            revision: git_revision_full().unwrap_or_default(),
```

Comment:
Sorry to pessimize, but is there anything that relied on it being the full string before this?

Reply:
Adding to this, what are the cases in which it can panic? Having the revision silently become `unknown` will just fail in even weirded ways further down the line, as it will try to use a branch called `unknown` instead of a revision etc

Reply:
> Sorry to pessimize, but is there anything that relied on it being the full string before this?

Wasn't this because of the git commit hash being cut short if it was proven by git to be unique based on the other commits in the tree?

It varied based on how many branches were pulled, so we got version mismatches on the commit hash being different lenghts depending on which commits were proven to be unique enough based on which commits were in the tree on the build VM